entproto/field.go

```diff
@@ -35,13 +36,29 @@ func Field(num int, options ...FieldOption) schema.Annotation {
 }
 
 type pbfield struct {
-	Number int
+	Number           int
+	OverrideType     descriptorpb.FieldDescriptorProto_Type
+	OverrideTypeName string
 }
 
 func (f pbfield) Name() string {
 	return FieldAnnotation
 }
 
+// OverrideType overrides the default mapping between ent types and protobuf types.
+func OverrideType(typ descriptorpb.FieldDescriptorProto_Type) FieldOption {
+	return func(p *pbfield) {
+		p.OverrideType = typ
+	}
+}
+
+// OverrideTypeName sets the pb descriptors type name, needed if the OverrideType attribute is TYPE_ENUM or TYPE_MESSAGE.
+func OverrideTypeName(n string) FieldOption {
+	return func(p *pbfield) {
+		p.OverrideTypeName = n
+	}
+}
```

Comment:
Looks good, but remove the `Override` string from this function. 

Also, consider adding a usage example in the godoc. I use it a lot. 

```suggestion
func Type(typ descriptorpb.FieldDescriptorProto_Type) FieldOption {
	return func(p *pbfield) {
		p.Type = typ
	}
}

// TypeName sets the pb descriptors type name, needed if the OverrideType attribute is TYPE_ENUM or TYPE_MESSAGE.
func TypeName(n string) FieldOption {
	return func(p *pbfield) {
		p.TypeName = n
	}
}
```